### PR TITLE
Feature775 update go

### LINKED ISSUE #650: go build fails with go1.16.4 because of missing go.mod file
## Situation
Executing integration tests with

```
./gradlew integrationtest
```
on my local machine fails with:
```
> Task :sechub-cli:buildGo FAILED
Build go:Start building package 'daimler.com/sechub'
>building:platform/linux-386
no required module provides package daimler.com/sechub: go.mod file not found in current directory or any parent directory; see 'go help modules'
Go build failed because of an error
```

Checking version:
```
> $ go version
go version go1.16.4 linux/amd64
```
## Wanted
Build shall work with newer go versions as well - e.g. for security reasons (compiler output

## Solutions
There a two possible options IMHO:

A) integrate adequate `go.mod` file which works for 1.15.x and also 1.16.x
B) Make a workaround via setting GOPATH ala 
    ```
    GOPATH="~/go:./sechub-cli" inside the gradle build file
    ```

### LINKED ISSUE #775: Unable to build with Go 1.16
# Problem

Unable to build with Go 1.16:

~~~
$ go version
go version go1.16.7 linux/amd64
~~~

Build error:

~~~
$ ./gradlew buildGo testGo

> Configure project :
BUILD versioning
- Server  :0.0.0 [0.0.0-28407d4-20210903162301]
- Client  :0.0.0 [0.0.0-28407d4-20210903162301]
- PDS     :0.0.0 [0.0.0-28407d4-20210903162301]
- Time elapsed for versioning:0.236 seconds

> Task :sechub-cli:genGo
generate 'constants_filepatterns_gen.go'

> Task :sechub-cli:buildGo FAILED
Build go:Start building package 'daimler.com/sechub'
>building:platform/linux-386
go tool compile: exit status 1
unsupported setting GO386=387. Consider using GO386=softfloat instead.
Go build failed because of an error

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':sechub-cli:buildGo'.
> Process 'command '/home/developer/dev/projects/sechub/sechub-cli/build_go.sh'' finished with non-zero exit value 1
~~~

The issue is, that Go 1.16 dropped support for `GO386=387` (see: https://github.com/golang/go/issues/40255). Now, SSE2 instructions are required to build Go for the Intel x86 32-bit architecture.

# Solution

Set the `GO386` environment variable to `softfloat` and build :

~~~
$ export GO386=softfloat
$ ./gradlew buildGo testGo
~~~

or for computers with SSE2 instructions. Only available on Pentium 4/Opteron/Athlon 64 or later:

~~~
$ export GO386=sse2
$ ./gradlew buildGo testGo
~~~

Extract from the [documentation](https://golang.org/doc/install/source):
 
~~~
* $GO386 (for 386 only, defaults to sse2)

  This variable controls how gc implements floating point computations.

    * GO386=softfloat: use software floating point operations; should support all x86 chips (Pentium MMX or later).
    * GO386=sse2: use SSE2 for floating point operations; has better performance but only available on Pentium 4/Opteron/Athlon 64 or later.
~~~

### PR DESCRIPTION
**Pull request by sven-dmlr**
enable Go versions > 1.15
- module handling added
- Go version update to 1.17

closes #650 
closes #775

### CODE REVIEW

File: .github/workflows/create-releases.yml [hunk 1]
```diff
@@ -53,7 +53,7 @@ jobs:
       - name: Set up Go
         uses: actions/setup-go@v2
         with:
-          go-version: 1.15.3
+          go-version: 1.17
```
Comment (Jeeppler) [inline .github/workflows/create-releases.yml:56]: The most current version is go-version: 1.17.6. Does the minor version matter? You used the minor version for 1.15.

**Review by Jeeppler — CHANGES_REQUESTED** [review]
See comments.

File: .github/workflows/create-releases.yml (hunk 1 above)
Comment (sven-dmlr) [reply .github/workflows/create-releases.yml:56]: `1.17` selects automatically the latest 1.17 release - so we have always the latest patches/fixes.
In Github Actions, you can see that currently go1.17.6 is used.

File: sechub-cli/build_go.sh [hunk 2]
```diff
@@ -19,16 +20,50 @@ if [[ -z "$platforms" ]]; then
...
+    echo "# Initialize go modules"
+    cd main/
+    go mod init $MOD_BASENAME
+    for i in $SUBMODULES ; do
+        pushd ../$i
+        go mod init $MOD_BASENAME/$i
+        popd >/dev/null
+        go mod edit -replace $MOD_BASENAME/$i=../$i
+    done
+    go mod tidy
```
[the remark this answers is not in the document]
Comment (sven-dmlr) [reply sechub-cli/build_go.sh]: Good point :+1: 
I added a comment explaining its purpose.

**Review by Jeeppler — APPROVED** [review]

**Review by cravity — APPROVED** [review]

**Pull request merged**